website_sale_hide_street2/views/website_sale_template.xml

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="address" inherit_id="website_sale.address">
+        <xpath expr="//div[contains(@t-attf-class, 'col-lg-12 div_street2')]" position="attributes">
```

Comment:
This seems to be right, but could you try an alternative non-dependent on class styles (not tested)?
```suggestion
        <xpath expr="//input[@name='street2']/parent::div" position="attributes">
```